### ISSUE TITLE
Widgets: RectangleSelector in Mpl 2.1.0 does not behave as described

### ISSUE THREAD

**Issue by alsast** [matplotlib/matplotlib#9608]
### Bug report

In Python 3.6.3 and the related matplotlib (just compiled from sources) the RectangleSelector does no longer behave as in Python 2.7 and as is described. Instead of returning the mouse click coordinates in the event variable eclick and the release coordinates in erelease it returns the lower left corner in eclick and the upper right corner in erelease. This "normalization" reduces the information content and, thus, the usability of the selector. Only using the old version a user can switch to drawmode='line' and use the selector to input a displacement vector.

**Comment by tacaswell** [comment]
What version of Matplotlib are you testing?  I would be very surprised if this changes is tied to the version of _python_.

**Comment by alsast** [comment]
Well, the python3 uses matplotlib 2.1.0 and there the RectangleSelector does not behave like described -- the python2 has matplotlib 1.3.1 installed, where the code reflects the description.

**Comment by tacaswell** [comment]
@alsast Sorry we broke this for you.

I suspect that this was a consequence of #3937 which went in for 1.5.  At a minimum we need to fix the docstring to reflect the current behavior and consider flipping it back or making it configurable.   I have a memory of another discussion of this, but can not find it quickly.

I think the normalization of the points made managing the resizing of selection region tractable? attn @blink1073

**Comment by blink1073** [comment]
The normalization was added because you might be dragging one of the handles, or the whole object.  It could be improved to preserve the order though so you don't lose information.

https://github.com/matplotlib/matplotlib/blob/58ec50606ffe62de827897b3ebd72a235d242ab1/lib/matplotlib/widgets.py#L2154-L2163

**Comment by blink1073** [comment]
Actually, it isn't just order.  What we could do is only normalize if the events aren't both part of the new corner points.

**Comment by blink1073** [comment]
Or we could revert entirely and you use the extents if you need the true corners.

**Comment by rowlesmr** [comment]
Can the `RectangleSelector` be made configurable to report either `extents` or the actual mouse press and release coordinates?

If this is a thing that people think is OK, I'll try and figure out how to write a patch for it.

**Comment by anntzer** [comment]
I would rather go with
> Or we could revert entirely and you use the extents if you need the true corners.

rather than adding more configurability...

**Comment by rowlesmr** [comment]
OK. I'll see if I can contribute: Make the behaviour for `onselect` for `RectangleSelector` match the docstring (again). If you need `extents`, then you can call that explicitly.

**Comment by rowlesmr** [comment]
I've submitted a PR for this issue: #20611

**Comment by tacaswell** [comment]
re-reading this again, I noticed

> Only using the old version a user can switch to drawmode='line' and use the selector to input a displacement vector.

which was a very clever use of `RectangleSelector` to get a `VectorSelector` even if we did not understand that we provided it!  @alsast Sorry for not understanding you correctly in 2017.

However, I think that at this point (~4 years later) it is too late to revert the change to `RectangleSelector` on the risk that we now have users who are relying on the new order.   I think the two actions we should take from this issue are 

1. update the documentation of RectangleSelector to make it match what the code does
2. implement a `VectorSelector` that is intended from selecting displacement vectors.  Now that we have a growing convention around putting handles on the selectors, I think it would make sense to be Line2D with handles on each and and one in the middle.
